[BUG]: PCSX2 failng to compile in linux

### Describe the Bug

Compiling fails with:
```
/home/testing/Desktop/pcsx2/pcsx2/pcsx2/GS/Renderers/HW/GSRendererHW.cpp:1215:50: error: cannot bind non-const lvalue reference of type ‘std::string&’ {aka ‘std::__cxx11::basic_string<char>&’} to an rvalue of type ‘std::__cxx11::basic_string<char>’
 1215 |                         DumpVertices(m_dump_root + s);
      |                                      ~~~~~~~~~~~~^~~
```

### Reproduction Steps

Attempt to compile PCSX2.

Test specs:

* CPU: Intel i7-10870H
* GPU: Nvidia GTX 3070M / driver 510.47
* OS: Linux Mint 20.3 / kernel 5.13
* Compiler: gcc 11.1.0
* Compile log: [here](https://pastebin.com/EphW6aT5)

### Expected Behavior

PCSX2 should compile.  The last known good compilation was 2f028c5f402dfe1a4b1f411d3df02b248e5ceb39.  Can bisect to the specific problem commit if necessary.

### PCSX2 Revision

bbe3af7f6144bfd2ff91ad40a71f94f9333cbe29

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Linux Mint 20.3

--- a/pcsx2/GS/GSState.cpp
+++ b/pcsx2/GS/GSState.cpp
@@ -552,7 +552,7 @@ float GSState::GetTvRefreshRate()
 	__assume(0); // unreachable
 }
 
-void GSState::DumpVertices(std::string& filename)
+void GSState::DumpVertices(const std::string& filename)
 {
 	std::ofstream file(filename);
 

--- a/pcsx2/GS/GSState.h
+++ b/pcsx2/GS/GSState.h
@@ -303,7 +303,7 @@ public:
 	void SetRegsMem(u8* basemem) { m_regs = reinterpret_cast<GSPrivRegSet*>(basemem); }
 
 	void SetFrameSkip(int skip);
-	void DumpVertices(std::string& filename);
+	void DumpVertices(const std::string& filename);
 
 	PRIM_OVERLAP PrimitiveOverlap();
 	GIFRegTEX0 GetTex0Layer(u32 lod);

